src/snowflake/snowpark/_internal/telemetry.py

```diff
@@ -428,6 +433,42 @@ def send_large_query_breakdown_telemetry(
         }
         self.send(message)
 
+    def send_post_compilation_stage_telemetry(
+        self,
+        session_id: int,
+        plan_uuid: str,
+        cte_optimization_enabled: bool,
```

Comment:
that seems a lot a arguments now, I think we can simply take a summary_value, which is a dict[str, any] here. in the future, we can add more data inside easily